worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/CreditStreamManager.java

```diff
@@ -158,11 +171,55 @@ private void addCredit(MapPartitionData mapPartitionData, int numCredit, long st
     }
   }
 
+  private void notifyRequiredSegment(
+      MapPartitionData mapPartitionData, int requiredSegmentId, long streamId, int subPartitionId) {
+    logger.debug(
+        "Receive RequiredSegment from client, streamId: {}, requiredSegmentId: {}, subPartitionId: {}",
+        streamId,
+        requiredSegmentId,
+        subPartitionId);
+    try {
+      if (mapPartitionData != null) {
+        checkState(mapPartitionData instanceof SegmentMapPartitionData);
```

Comment:
Should this use `checkState`  to check whether `mapPartitionData instanceof SegmentMapPartitionData`? IMO, we could just move the check in if condition.